sqlbrite-lint/src/test/java/com/squareup/sqlbrite3/SqlBriteArgCountDetectorTest.kt

```diff
@@ -0,0 +1,203 @@
+
+              public class Test {
+
+                  private static final String QUERY = "SELECT name FROM table WHERE id = ?";
+
+                  public void test() {
+                    BriteDatabase db = new BriteDatabase();
+                    db.query(QUERY, "id");
+                  }
+
+              }
+            """.trimIndent()))
+        .run()
+        .expectClean()
+  }
+
+  fun testCleanCaseWithQueryAsBinaryExpression() {
+    lint().files(
+        BRITE_DATABASE_STUB,
+        java(
+            """
+              package test.pkg;
+
+              import com.squareup.sqlbrite3.BriteDatabase;
+
+              public class Test {
+
+                  private static final String QUERY = "SELECT name FROM table WHERE ";
+
+                  public void test() {
+                    BriteDatabase db = new BriteDatabase();
+                    db.query(QUERY + "id = ?", "id");
+                  }
+
+              }
+            """.trimIndent()))
+        .run()
+        .expectClean()
+  }
+
+  fun testCleanCaseWithQueryThatCantBeEvaluated() {
+    lint().files(
+        BRITE_DATABASE_STUB,
+        java(
+            """
+              package test.pkg;
+
+              import com.squareup.sqlbrite3.BriteDatabase;
+
+              public class Test {
+
+                  private static final String QUERY = "SELECT name FROM table WHERE id = ?";
+
+                  public void test() {
+                    BriteDatabase db = new BriteDatabase();
+                    db.query(query(), "id");
+                  }
+
+                  private String query() {
+                    return QUERY + " age = ?";
+                  }
+
+              }
+            """.trimIndent()))
+        .run()
+        .expectClean()
+  }
+
+  fun testCleanCaseWithNonVarargMethodCall() {
+    lint().files(
+        BRITE_DATABASE_STUB,
+        java(
+            """
+              package test.pkg;
+
+              import com.squareup.sqlbrite3.BriteDatabase;
+
+              public class Test {
+
+                  public void test() {
+                    BriteDatabase db = new BriteDatabase();
+                    db.createQuery("table", 42);
+                  }
+
+              }
+            """.trimIndent()))
+        .run()
+        .expectClean()
+  }
+
+  fun testQueryMethodWithWrongNumberOfArguments() {
+    lint().files(
+        BRITE_DATABASE_STUB,
+        java(
+            """
+              package test.pkg;
+
+              import com.squareup.sqlbrite3.BriteDatabase;
+
+              public class Test {
+
+                  private static final String QUERY = "SELECT name FROM table WHERE id = ?";
+
+                  public void test() {
+                    BriteDatabase db = new BriteDatabase();
+                    db.query(QUERY);
+                  }
+
+              }
+            """.trimIndent()))
+        .run()
+        .expect("src/test/pkg/Test.java:11: " +
```

Comment:
You could use the """ here with trimMargin() similiar to:

```kotlin
      .expect("""
          |res/layout/random.xml: Warning: Layout does not start with one of the following prefixes: activity_, view_, fragment_, dialog_, bottom_sheet_, adapter_item_, divider_, space_ [WrongLayoutName]
          |0 errors, 1 warnings
          """.trimMargin())
```

Reply:
I don't know, I think that I got an idea how the function works, but I just can't make it work. Tbh, not even sure if it brings more readability than regular string concatenation in that case